metagraph/src/common/vectors/sd_vector_disk.hpp

```diff
@@ -0,0 +1,530 @@
+#ifndef __SD_VECTOR_DISK__
+#define __SD_VECTOR_DISK__
+
+#include <type_traits>
+
+#include <sdsl/sd_vector.hpp>
+#include <sdsl/int_vector.hpp>
+#include <sdsl/int_vector_buffer.hpp>
+#include <sdsl/select_support_mcl.hpp>
+#include <sdsl/util.hpp>
+#include <sdsl/iterators.hpp>
+
+// Based on https://github.com/simongog/sdsl-lite/blob/master/include/sdsl/sd_vector.hpp
+
+// FYI: This implementation does NOT support multithreading
+
+namespace sdsl {
+
+// forward declaration needed for friend declaration
+template<uint8_t t_b          = 1,
+         class t_hi_bit_vector= bit_vector,
+         class t_select_1     = typename t_hi_bit_vector::select_1_type,
+         class t_select_0     = typename t_hi_bit_vector::select_0_type>
+class rank_support_sd_disk;  // in sd_vector
+
+// forward declaration needed for friend declaration
+template<uint8_t t_b          = 1,
+         class t_hi_bit_vector= bit_vector,
+         class t_select_1     = typename t_hi_bit_vector::select_1_type,
+         class t_select_0     = typename t_hi_bit_vector::select_0_type>
+class select_support_sd_disk;  // in sd_vector
```

Comment:
how much would it break things if we make the storage type of `m_low` and `m_high` template parameters in `sdsl`? we don't have to do this here, but keep this in mind for the future.

Reply:
`m_high` is already a template (`t_hi_bit_vector`). Making `m_low` a template doesn't make much sense, there is too much code specific to `int_vector_buffer`. So a good way would be to make a parent class and move all shared code there but for this, we would need to rewrite sd_vector in sdsl.